### PR TITLE
Update DG - Find command

### PR DESCRIPTION
**Pull request by jonahtanjz**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -270,21 +270,58 @@ The constraints above have been applied after careful consideration of the needs
...
     * Pros: Improves OOP aspect of the code.
     * Cons: Unnecessarily complicates the code.
 
+### Find feature
+
+The find feature allows users to search for food items based on description, expiration date, priority and/or tags.
+
+#### Implementation:
+
+The find feature is achieved by setting a `predicate` on the `filteredList` located in `ModelManager`. This can be done using the method `ModelManager#updateFilteredFoodList(Predicate<Food> predicate)`, which uses the parameter as the `predicate`. There are 4 valid `predicate`, namely `DescriptionContainsKeywordsPredicate`, `ExpirySearchPredicate`, `PrioritySearchPredicate` and `TagSearchPredicate`, all of which implements the `Predicate<Food>` interface. Based on the user's `find` command, each of the `predicate` is generated with the relevant search parameters. Then, the `FindCommand#combinePredicates()` method will combine all the predicates into a single `predicate` and is passed to `ModelManager#updateFilteredFoodList(Predicate<Food> predicate)`.
```
Comment (rolandyuwy) [inline docs/DeveloperGuide.md]: We should probably decide to standard formatting for grammar here. Would "4 valid predicates" be better than "4 valid `predicate`"? Or `predicate` objects?

Comment (rolandyuwy) [inline docs/DeveloperGuide.md]: I think it should be "all of which implement"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -270,21 +270,58 @@ The constraints above have been applied after careful consideration of the needs
...
+
+The `find` command uses `ArgumentMultimap` to get the parameters of each `prefix`, similar to the `add` command. The `find` command needs to have at least one `prefix` present. If a `prefix` is present, then its respective `predicate` will be generated.
+
+The following is the class diagram for the Find feature:
+
+![FindClasseDiagram](images/FindClassDiagram.png)
+
+#### Implementation Rationale:
+
+Since the user can search for food items based on either the description, expiration date, priority or tags, the `find` command should allow searching for 1 or more of the above combination. This increases the flexibility in the `find` command, which allows the user to define the specificity of their search.
```
Comment (rolandyuwy) [inline docs/DeveloperGuide.md]: Let's standardize '1' to 'one'.

Comment (rolandyuwy) [reply docs/DeveloperGuide.md]: Similarly for Alternative section

**Review by rolandyuwy — COMMENTED** [review]
Minor suggestions, otherwise LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (jonahtanjz) [reply docs/DeveloperGuide.md]: Ok. I think I will change to four valid predicates.

Comment (jonahtanjz) [reply docs/DeveloperGuide.md]: Noted. Will change.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (jonahtanjz) [reply docs/DeveloperGuide.md]: Ok. Will change these as well as the '4' to 'four' for the predicates.

**Comment by rolandyuwy** [comment]
LGTM!

**Pull request merged**